java/core/src/test/java/sleeper/core/range/RangeTest.java

```diff
@@ -30,643 +33,782 @@
+        public void testEqualsForCanonicallyIntType() {
+            // Given
+            Field field = new Field("key", new IntType());
+            Schema schema = Schema.builder().rowKeyFields(field).build();
+            RangeFactory rangeFactory = new RangeFactory(schema);
+            Range range1 = rangeFactory.createRange(field, 10, true, 20, true);
+            Range range2 = rangeFactory.createRange(field, 10, false, 20, false);
+            Range range3 = rangeFactory.createRange(field, 9, false, 21, false);
+            Range range4 = rangeFactory.createRange(field, 9, false, 20, true);
+            Range range5 = rangeFactory.createRange(field, 10, true, 21, false);
+
+            // When / Then
+            assertThat(range1.equalsCanonicalised(range2)).isFalse();
+            assertThat(range1.equalsCanonicalised(range3)).isTrue();
+            assertThat(range1.equalsCanonicalised(range4)).isTrue();
+            assertThat(range1.equalsCanonicalised(range5)).isTrue();
+        }
+
+        @Test
+        void shouldEqualsForCanonicallyLongType() {
+            // Given
+            Field field = new Field("key", new LongType());
+            Schema schema = Schema.builder().rowKeyFields(field).build();
+            RangeFactory rangeFactory = new RangeFactory(schema);
+            Range range1 = rangeFactory.createRange(field, 5L, true, 9L, true);
+            Range range2 = rangeFactory.createRange(field, 5L, false, 9L, false);
+            Range range3 = rangeFactory.createRange(field, 4L, false, 10L, false);
+            Range range4 = rangeFactory.createRange(field, 4L, false, 9L, true);
+            Range range5 = rangeFactory.createRange(field, 5L, true, 10L, false);
+
+            // When / Then
+            assertThat(range1.equalsCanonicalised(range2)).isFalse();
+            assertThat(range1.equalsCanonicalised(range3)).isTrue();
+            assertThat(range1.equalsCanonicalised(range4)).isTrue();
+            assertThat(range1.equalsCanonicalised(range5)).isTrue();
+        }
+
+        @Test
+        void shouldEqualsForCanonicallyByteArrayType() {
+            // Given
+            Field field = new Field("key", new ByteArrayType());
+            Schema schema = Schema.builder().rowKeyFields(field).build();
+            RangeFactory rangeFactory = new RangeFactory(schema);
+
+            Range range1 = rangeFactory.createRange(field, nextByteArrayValue(new byte[]{10, 10}), true, new byte[]{20, 20}, true);
+            Range range2 = rangeFactory.createRange(field, new byte[]{10, 10}, false, new byte[]{20, 20}, false);
+            Range range3 = rangeFactory.createRange(field, new byte[]{10, 10}, false, nextByteArrayValue(new byte[]{20, 20}), false);
+            Range range4 = rangeFactory.createRange(field, new byte[]{10, 10}, false, new byte[]{20, 20}, true);
+            Range range5 = rangeFactory.createRange(field, nextByteArrayValue(new byte[]{10, 10}), true, nextByteArrayValue(new byte[]{20, 20}), false);
```

Comment:
The nextByteArrayValue calls duplicate the production implementation for canonicalising a range. Please put the actual value here explicitly, like you have for the other types. It doesn't look like the nextByteArrayValue method is needed.